### PR TITLE
FIX: Fixes #26 Use config system to enable/disable FB cache-clearing

### LINKED ISSUE #26: Facebook scrape ping gets triggerd on build
On builds the scrape ping get's triggerd, on a site with many pages i get errors like "Client error 403 Forbidden, Application request limit reached". 

I haven't gotten around checking why this get's triggered on build, maybe it isn't that important on a live site as the error is soft. But on a dev site it's pretty annoying, maybe adding `if (!Director::isDev())` would suffice.

### PR DESCRIPTION
**Pull request by phptek**

### CODE REVIEW

**Comment by phptek** [comment]
Not sure why GIT thinks 14 files have changed. I set "filemode = false" in .git/config...

File: src/ShareCare.php [hunk 1]
```diff
@@ -49,6 +49,14 @@ class ShareCare extends DataExtension
      */
     private static $pinterest = false;
 
+    /**
+     * Allow userland config to conditionally permit cache-clearing for Facebook
+     * after every page-write.
+     *
+     * @var bool
+     */
+    private static $enable_facebook_cache_clear = false;
```
Comment (jonom) [inline src/ShareCare.php]: This needs to have a default value of true for backwards compatibility

File: _config/ShareCare.yml [hunk 2]
```diff
@@ -9,4 +9,4 @@ Page:
     # - JonoM\ShareCare\ShareCareSingleSummary
 # JonoM\ShareCare\ShareCare:
 #   pinterest: true
-#   twitter_username: 'test-user'
+#   twitter_username: 'test-user'
```
Comment (jonom) [inline _config/ShareCare.yml]: Remove this change please (it's harmless but doesn't need to be here)

File: README.md [hunk 3]
```diff
@@ -137,6 +137,22 @@ JonoM\ShareCare\ShareCare:
...
 if this feature is disabled.
 
+### Control Facebook's Cache
+
+Out of the box, this module will attempt to clear Facebook's knowledge of your site
+or app, after every page write. Set the following config to allow a level of control
+over which environment this should be run on.
+
+```yml
+Except:
```
Comment (jonom) [inline README.md]: Please change to:

```yml
JonoM\ShareCare\ShareCare:
  enable_facebook_cache_clear: false
---
Only:
  environment:
    - live
---
JonoM\ShareCare\ShareCare:
  enable_facebook_cache_clear: true
```

**Review by jonom — CHANGES_REQUESTED** [review]
@phptek thanks for this! Maybe your `filemode = false` needs to be `fileMode = false`. If that doesn't work can you just change the permissions back to 100644 for affected files?

**Comment by phptek** [comment]
Fixed-up the commit:

* Applied suggested edits
* Removed permission changes
* Force-pushed the feature branch

**Comment by phptek** [comment]
@jonom any update on this? Thanks :-)

**Comment by jonom** [comment]
Sorry for the delay @phptek! I've now tested and tagged 4.1 after tweaking the docs a little. Thanks for your contribution!

**Pull request merged**